src/v3/src/transformer/field/transform.ts

```diff
@@ -91,12 +91,18 @@ export const transformStepInputs = (
       const uischema = mapUiElement(input);
       acc.uischema.elements = [...acc.uischema.elements, uischema];
 
+      if (type === 'boolean' && required) {
+        acc.data[name] = Boolean(input.value);
+      }
+
       // add client validation for "required" field
       // do not validate immutable fields, they will always be added to payload programatically
       if (required && mutable !== false && type !== 'object') {
         acc.dataSchema[name] = {
           validate(data) {
-            const isValid = !!data[name];
+            // in the case of a required boolean input, just return true
+            // if the backend requires the checbox val to be true, error will be displayed
+            const isValid = typeof data[name] === 'boolean' ? true : !!data[name];
```

Comment:
Is there any way to combine this logic with the next line using early return? e.g.
```js
// i'm thinking you might start with this but haven't thought through how to handle the right side of the ternary
if (typeof data[name] === 'boolean') return undefined;
```